### DISCUSSION TITLE
MODBUS Commands in Artisan Events

### DISCUSSION THREAD

**Discussion by masaeb28** [artisan-roaster-scope/artisan#1767] [Q&A]
Does anyone use modbus commands to control burner output % via event buttons or scale? I’d like to pick someone’s brain about modbus and communicating with my Watlow EZ Zone PD controller on my Ambex DY5 - I’m getting headaches. 

I have the controller manual and I have artisan receiving BT and ET input values but I just don’t understand how to use artisan to write to controller registers to operate the burners. Hand holding is appreciated!

**Comment by roasterdave** [comment]
The Ambex DY5 is not a model number familiar to me.  First question then, is your roaster burner valve capable of being controlled from the Watlow, or is the burner valve manual?   If the valve is controllable do you know the make and model and which Watlow output it is connect to?

**Reply by masaeb28** [reply]
Sorry, mistype, it’s a YM5. Yes the roaster burner is capable of being controlled from the watlow. The roaster currently runs using Roaster Dynamics software which controls via PID or manual in software gas adjustments. 

Modulator is a Maxitrol Selectra M420H.

**Reply by roasterdave** [reply]
Great.  Now the unknown is which Watlow register is being used for the control loop.  Ambex has not provided [support to Artisan](https://artisan-scope.org/machines/index) and we don't know how your machine is configured.  You will help the Artisan community by posting back the correct information when you get it to work. 

Look in your Watlow manual  for the Control Loop Manual Power register.  The correct address depends on the Instance and Map actually used in your roaster.  You can try the register addresses shown below as a starting point, assuming yours is an EZ-Zone PM model.

Setup your Burner slider to write to the Control Loop Manual Power register (assuming address 2642).

![image](https://github.com/user-attachments/assets/94775e1e-d72a-45a1-b36a-e8e940895a79)

You will need to set the Control Loop Control Mode to manual (assuming address 2360).  In this context, manual means the Watlow is taking the % value from commands sent by Artisan. 

![image](https://github.com/user-attachments/assets/db1535e8-021a-4c10-8b10-67776cee3941)

Let us know how it goes.
